File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLImportedContact.java

package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.TLContext;
import com.github.badoualy.telegram.tl.core.TLObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readInt;
import static com.github.badoualy.telegram.tl.StreamUtils.readLong;
import static com.github.badoualy.telegram.tl.StreamUtils.writeInt;
import static com.github.badoualy.telegram.tl.StreamUtils.writeLong;


public class TLImportedContact extends TLObject {

    public static final int CLASS_ID = 0xd0028438;

    public TLImportedContact() {

    }


    public TLImportedContact(        int _userId,         long _clientId) {
        this.userId = _userId;
        this.clientId = _clientId;

    }


    public int getClassId() {
        return CLASS_ID;
    }


    protected int userId;

    protected long clientId;


    public int getUserId() {
        return userId;
    }

    public void setUserId(int value) {
        this.userId = value;
    }

    public long getClientId() {
        return clientId;
    }

    public void setClientId(long value) {
        this.clientId = value;
    }


    @Override
    public void serializeBody(OutputStream stream) throws IOException {

        writeInt(this.userId, stream);
        writeLong(this.clientId, stream);
    }


    @Override
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {

        this.userId = readInt(stream);
        this.clientId = readLong(stream);
    }


    @Override
    public String toString() {
        return "importedContact#d0028438";
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLImportedContact.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.core.TLObject
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException

/**
 * importedContact#d0028438
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLImportedContact() : TLObject() {
    var userId: Int = 0

    var clientId: Long = 0L

    private val _constructor: String = "importedContact#d0028438"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(userId: Int, clientId: Long) : this() {
        this.userId = userId
        this.clientId = clientId
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeInt(userId)
        writeLong(clientId)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        userId = readInt()
        clientId = readLong()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += SIZE_INT32
        size += SIZE_INT64
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLImportedContact) return false
        if (other === this) return true

        return userId == other.userId
                && clientId == other.clientId
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xd0028438.toInt()
    }
}
